### PR TITLE
[BEAM-4776] Add MetricQueryResults.allMetrics() helper

### PR DESCRIPTION
**Pull request by ryan-williams**
(factored out of #7823)

Simple helper for the common case of fetching all metrics from a `MetricQueryResults`, replacing both `queryResults(null)` and `queryMetrics(MetricsFilter.builder().build()`

R: @robertwb, @ajamato 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](../.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ryan-williams** [comment]
[seems like a flaky test](https://scans.gradle.com/s/m2vawlhlp25ze/console-log?task=:beam-sdks-java-io-cassandra:test#L6):

```

org.apache.beam.sdk.io.cassandra.CassandraIOTest > classMethod FAILED
--
java.lang.AssertionError at CassandraIOTest.java:100
18:50:38.650 [StorageServiceShutdownHook] INFO  o.a.cassandra.hints.HintsService - Paused hints dispatch
 
org.apache.beam.sdk.io.cassandra.CassandraIOTest > classMethod FAILED
java.lang.NullPointerException at CassandraIOTest.java:125
```

**Comment by ryan-williams** [comment]
Run Java PreCommit

**Comment by ryan-williams** [comment]
Run Portable_Python PreCommit

**Comment by ryan-williams** [comment]
[Portable_Python PreCommit OOM'd!](https://scans.gradle.com/s/mxqgo6act6ur6/console-log?task=:beam-sdks-python:portableWordCountStreaming#L1090):

```
[flink-runner-job-server] ERROR org.apache.beam.runners.flink.FlinkJobInvocation - Error during job invocation BeamApp-root-0223053928-e1bf7e3d_a7784282-debe-4f1c-a18e-9edb11c7aec5.
--
org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
at org.apache.flink.runtime.jobmaster.JobResult.toJobExecutionResult(JobResult.java:146)
at org.apache.flink.runtime.minicluster.MiniCluster.executeJobBlocking(MiniCluster.java:632)
at org.apache.flink.streaming.api.environment.LocalStreamEnvironment.execute(LocalStreamEnvironment.java:123)
at org.apache.beam.runners.flink.FlinkStreamingPortablePipelineTranslator$StreamingTranslationContext.execute(FlinkStreamingPortablePipelineTranslator.java:158)
at org.apache.beam.runners.flink.FlinkJobInvocation.runPipelineWithTranslator(FlinkJobInvocation.java:149)
at org.apache.beam.runners.flink.FlinkJobInvocation.runPipeline(FlinkJobInvocation.java:114)
at org.apache.beam.vendor.guava.v20_0.com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
at org.apache.beam.vendor.guava.v20_0.com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
at org.apache.beam.vendor.guava.v20_0.com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.OutOfMemoryError: unable to create new native thread
at java.lang.Thread.start0(Native Method)
at java.lang.Thread.start(Thread.java:717)
at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:957)
at java.util.concurrent.ThreadPoolExecutor.ensurePrestart(ThreadPoolExecutor.java:1603)
 
at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:334)
at java.util.concurrent.ScheduledThreadPoolExecutor.scheduleAtFixedRate(ScheduledThreadPoolExecutor.java:573)
at org.apache.flink.streaming.runtime.tasks.SystemProcessingTimeService.scheduleAtFixedRate(SystemProcessingTimeService.java:146)
at org.apache.beam.runners.flink.translation.wrappers.streaming.DoFnOperator.open(DoFnOperator.java:378)
at org.apache.beam.runners.flink.translation.wrappers.streaming.ExecutableStageDoFnOperator.open(ExecutableStageDoFnOperator.java:203)
at org.apache.flink.streaming.runtime.tasks.StreamTask.openAllOperators(StreamTask.java:420)
at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:296)
at org.apache.flink.runtime.taskmanager.Task.run(Task.java:712)
```

File: .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Flink.groovy [hunk 1]
```diff
@@ -31,7 +31,7 @@ PostcommitJobBuilder.postCommitJob('beam_PostCommit_Java_ValidatesRunner_Flink',
   // Publish all test results to Jenkins
   publishers {
     archiveJunit('**/build/test-results/**/*.xml')
-  }
+  }metconstep
```
Comment (angoenka) [inline .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Flink.groovy]: Is this needed?

Comment (ryan-williams) [reply .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Flink.groovy]: oh, that's a typo! thanks, let me fix now

Comment (ryan-williams) [reply .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Flink.groovy]: pushed a corrected version, good catch!

that's me trying to search for the `MetricsContainerStepMap` file when IntelliJ is not ready, and has a race condition with the file-search popup appearing and the text going there 😞

**Pull request merged**